### PR TITLE
Switch factories to strings instead of constants

### PR DESCRIPTION
**Pull request by cbrunsdon**
When we define the factories we don't actually care if the constants are
resolved to real objects or not

### CODE REVIEW

File: core/spec/lib/spree/core/validators/email_spec.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+require 'spree/core/validators/email'
```
Comment (kennyadsl) [inline core/spec/lib/spree/core/validators/email_spec.rb]: is this supposed to stay here?

**Comment by mtomov** [comment]
Say again, what is the advantage of using strings? 

I was thinking that using real objects sort of adds a validation check that there really exists this object, which I thought might be good.

**Comment by tvdeyen** [comment]
One of the advantages is that the constant doesn't have to be loaded when loading the factories, only when you actually use it.

I get the validation argument and first thought the same, but as our own test suite uses all of these factories we have confidence that they are valid ;)

**Comment by mtomov** [comment]
Makes sense, thanks!

File: core/spec/lib/spree/core/validators/email_spec.rb (hunk 1 above)
Comment (cbrunsdon) [reply core/spec/lib/spree/core/validators/email_spec.rb]: I split it into another commit. Long story short, without the order const loaded, EmailValidator isn't properly loaded and now that the order const isn't loaded when its factory is, this isn't required anymore.

**Comment by kennyadsl** [comment]
For the validation argument, we also have [specs for factories](https://github.com/solidusio/solidus/tree/3f485cf0cddf37085183a5d55976712d011b2e30/core/spec/lib/spree/core/testing_support/factories).

**Pull request merged**